fibonacci.js

```diff
@@ -0,0 +1,123 @@
+// Variables:
+const fibInput = document.getElementById("form12");
+const buttonLocal = document.getElementById("button-local");
+const buttonServer = document.getElementById("button-server");
+const fibNumber = document.querySelector("#fib-number");
+const spinner = document.getElementById("spinner");
+const error50 = document.getElementById("error-50");
+const resultLog = document.getElementById("result-line");
+const resultsSpinner = document.getElementById("spinner-results");
+const checkBox = document.getElementById("flexCheckDefault");
+const error00 = document.getElementById("error-00");
+
+// Event listeners:
+window.addEventListener("load", getResultsList);
+
+fibInput.addEventListener("focus", clearInput);
+
+buttonServer.addEventListener("click", getFibNumber);
+
+buttonLocal.addEventListener("click", () => {
+  fibNumber.innerText = fibFunction(parseInt(fibInput.value));
+});
+
+// Swapping between local and server buttons
+checkBox.addEventListener("change", () => {
+  if (checkBox.checked) {
+    // hide the local button
+    buttonLocal.classList.add("d-none");
+    buttonServer.classList.remove("d-none");
+  } else if (!checkBox.checked) {
+    // hide the server button
+    buttonLocal.classList.remove("d-none");
+    buttonServer.classList.add("d-none");
+  }
+});
+
+// Function to calculate the Fibonacci number LOCALLY:
+function fibFunction(n) {
+  if (n < 0) {
+    spinner.classList.remove("d-none");
+    error00.classList.remove("d-none");
+    fibNumber.classList.add("d-none");
+    spinner.classList.add("d-none");
+  } else if (n < 2) {
+    spinner.classList.remove("d-none");
+    error00.classList.add("d-none");
+    spinner.classList.add("d-none");
+    return n;
+  } else if (n > 50) {
+    spinner.classList.remove("d-none");
+    error00.classList.add("d-none");
+    error50.classList.remove("d-none");
+    fibNumber.classList.add("d-none");
+    spinner.classList.add("d-none");
+  } else {
+    spinner.classList.remove("d-none");
+    error00.classList.add("d-none");
+    fibNumber.classList.remove("d-none");
+    spinner.classList.add("d-none");
+    return fibFunction(n - 1) + fibFunction(n - 2);
+  }
+}
+
+// Function to calculate the Fibonacci number ON THE SERVER:
+function getFibNumber() {
```

Comment:
Same very long try to break this down to a couple of function for more readability